src/createServer.js

```diff
@@ -1,8 +1,54 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'text/plain');
+
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+    if (!req.url.includes('/file')) {
+      res.statusCode = 400;
+      res.end('Invalid request. Not allowed traversal paths');
+
+      return;
+    }
+
+    if (!req.url.startsWith('/file/')) {
+      res.statusCode = 200;
+      res.end('Invalid request. Please start your path with "/file/"');
+
+      return;
+    }
+
+    if (req.url.includes('//')) {
+      res.statusCode = 404;
+      res.end('File not found');
+
+      return;
+    }
+
+    const filename = normalizedUrl.pathname.slice('/file/'.length)
+      || 'index.html';
+
+    fs.readFile(`./public/${filename}`, (err, data) => {
+      if (!err) {
+        res.statusCode = 200;
```

Comment:
Tip* -  you can create object ( in ts - enum) - STATUS_CODES or something like that)